datafusion/functions/src/string/uuid.rs

```diff
@@ -87,7 +88,13 @@ impl ScalarUDFImpl for UuidFunc {
         if !args.is_empty() {
             return internal_err!("{} function does not accept arguments", self.name());
         }
-        let values = std::iter::repeat_with(|| Uuid::new_v4().to_string()).take(num_rows);
+
+        // Generate random u128 values
+        let mut rng = rand::thread_rng();
+        let mut randoms = vec![0u128; num_rows];
+        rng.fill(&mut randoms[..]);
+
+        let values = randoms.into_iter().map(|x| Uuid::from_u128(x).to_string());
```

Comment:
Thanks @simonvandel 
I'm just curious how this is faster as under the hood it uses the same approach
```
    pub fn new_v4() -> Uuid {
        // This is an optimized method for generating random UUIDs that just masks
        // out the bits for the version and variant and sets them both together
        Uuid::from_u128(
            crate::rng::u128() & 0xFFFFFFFFFFFF4FFFBFFFFFFFFFFFFFFF | 0x40008000000000000000,
        )
    }
```

probably `repeat_with().take()` is slow

Reply:
I tried calling `Uuid::new_v4()` instead of `Uuid::from_u128` with thread_rng. The performance was the same as before this PR.

I think the performance comes from not having to repeatedly call the rng for bytes. Instead, num_rows*36 bytes can be rng'ed at once.